### ISSUE TITLE
Issue with Guided Topic modeling

### ISSUE THREAD

**Issue by RamSingh1988** [MaartenGr/BERTopic#692]
Getting error while running guided topic modeling. I think the error occurs when there is no semantic matching available.

Error is in function _guided_topic_modeling

topic_model = BERTopic(language='english'
                ,seed_topic_list=topic_list,embedding_model=embedding_model
                ,umap_model= None
                ,hdbscan_model=None 
                ,verbose=True
                )

topics , prob = topic_model .fit_transform(X)

embeddings = topic_model._extract_embeddings(X,method="document", verbose=1) # X is document

seed_topic_embeddings = np.vstack([seed_topic_embeddings, embeddings.mean(axis=0)])

sim_matrix = cosine_similarity(embeddings, seed_topic_embeddings)

y = [np.argmax(sim_matrix[index]) for index in range(sim_matrix.shape[0])]

y = [val if val != len(seed_topic_list) else -1 for val in y] # this will be -1, when all the seed topic embeddings are highly similar to mean of document embeddings.

for seed_topic in range(len(seed_topic_list)):
    indices = [index for index, topic in enumerate(y) if topic == seed_topic]
    embeddings[indices] = np.average([embeddings[indices], seed_topic_embeddings[seed_topic]], weights=[3, 1]) # this is throwing error because indices are empty.

**Comment by MaartenGr** [comment]
Could you share a print screen of the error that you are getting? It helps me identify what the exact error is but also the entire error message. Also, from your message, it is not entirely clear which parts are code that you created yourself and which parts are code from internal BERTopic code. Could you use [fenced code blocks](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks) to make the code you shared a bit more clear?

**Comment by RamSingh1988** [comment]
Sorry for the previous code.

Below is the code that I am trying

```python
import pandas as pd
from bertopic import BERTopic


topic_list=[['worker', 'driver', 'nurse', 'doctor', 'staff', 'professor', 'labour'],
                                  ['trade', 'oil', 'crude', 'barrel', 'gas', 'import', 'energy']]

data = pd.read_csv('data.csv')

tp_model = BERTopic(language='english'
                    ,seed_topic_list=topic_list,embedding_model="all-MiniLM-L6-v2"
                    ,verbose=True
                     )

topics, probs = tp_model.fit_transform(documents =data.text)


````

![image](https://user-images.githubusercontent.com/10007363/187334571-c1e86705-59bf-48da-b0c5-3062eba1a19f.png)


Below is the code from your actual script where the error is occurring
``` python

y = [val if val != len(seed_topic_list) else -1 for val in y] # <- all the value of y is -1


for seed_topic in range(len(seed_topic_list)):
    indices = [index for index, topic in enumerate(y) if topic == seed_topic] # <- this is empty
    embeddings[indices] = np.average([embeddings[indices], seed_topic_embeddings[seed_topic]], weights=[3, 1]) #<- throwing error

````

**Comment by MaartenGr** [comment]
Based on the error that you provided, I would advise reading through the issue [here](https://github.com/MaartenGr/BERTopic/issues/458) where another user experienced the same issue. You can find some tips there that hopefully solves your issue.

**Comment by MaartenGr** [comment]
Due to inactivity, I'll be closing this for now. If you have any questions or want to continue the discussion, I'll make sure to re-open the issue!
